seqr/views/utils/dataset_utils.py

```diff
@@ -417,12 +416,11 @@ def _load_rna_seq(model_cls, file_path, user, mapping_file, ignore_extra_samples
     )
 
     warnings = []
-    if validate_samples:
-        samples = validate_samples(samples, sample_id_to_tissue_type, warnings)
 
     # Delete old data
     individual_db_ids = {s.individual_id for s in samples}
-    to_delete = model_cls.objects.filter(sample__individual_id__in=individual_db_ids).exclude(sample__data_source=data_source)
+    filters = {'sample__individual_id__in': individual_db_ids} if model_cls == RnaSeqOutlier else {'sample__in': samples}
```

Comment:
why would the behavior be different for different classes?

Reply:
Oh, yes. I should have added comments for this. I was thinking about the existing outlier data loaded to the sample models with tissue types. If we handle them like TPM data, these existing outlier data will be retained when loading new data.
We could manually delete the existing data or make the code backward compatible. What do you think?